src/main/java/com/nha/abdm/wrapper/hrp/serviceImpl/PatientTableService.java

```diff
@@ -42,29 +43,57 @@ public PatientTableService(PatientRepo patientRepo) {
         this.patientRepo = patientRepo;
     }
 
-    public List<CareContextBuilder> getCareContexts(String abhaAddress, Object content) {
-        if(Objects.nonNull(abhaAddress)){
-            Patients existingRecord = this.patientRepo.findByAbhaAddress(abhaAddress);
-            return existingRecord != null ? existingRecord.getCareContexts() : null;
-        }
-        if(content==DiscoverResponse.class){
-            DiscoverResponse data=(DiscoverResponse) content;
-            String patientIdentifier=null;
-            if(data.getPatient().getUnverifiedIdentifiers()!=null) patientIdentifier=data.getPatient().getUnverifiedIdentifiers().get(0).getValue();
-            String patientName=data.getPatient().getName();
-            String patientMobile=data.getPatient().getVerifiedIdentifiers().get(0).getValue();
-            if(Objects.nonNull(patientIdentifier)){
-                Patients existingRecord = this.patientRepo.findByPatientReference(patientIdentifier);
-                return existingRecord != null ? existingRecord.getCareContexts() : null;
+    @Transactional
+    public String addPatient(Object content) {
+        if(content.getClass()== LinkRecordsResponse.class){
+            LinkRecordsResponse data=(LinkRecordsResponse) content;
+            String patientReference=data.getPatientReference();
+            try{
+                Patients existingRecord = this.patientRepo.findByPatientReference(patientReference);
+                if (existingRecord == null) {
+                    return null;
```

Comment:
Throw an exception here saying `Patient not found`

Reply:
Throwing new RuntimeException("Patient not found")